plugin/evm/orderbook/contract_events_processor.go

```diff
@@ -386,6 +387,111 @@ func (cep *ContractEventsProcessor) handleClearingHouseEvent(event *types.Log) {
 	}
 }
 
+type TraderEvent struct {
+	Trader      common.Address
+	OrderId     common.Hash
+	OrderType   string
+	Removed     bool
+	EventName   string
+	Args        map[string]interface{}
+	BlockNumber *big.Int
+	BlockStatus string
+	Timestamp   interface{}
+}
+
+func (cep *ContractEventsProcessor) ProcessTraderEvents(events []*types.Log, blockStatus string) {
+	for _, event := range events {
+		removed := event.Removed
+		args := map[string]interface{}{}
+		eventName := ""
+		var orderId common.Hash
+		var orderType string
+		var trader common.Address
+		if len(event.Topics) > 1 {
+			trader = getAddressFromTopicHash(event.Topics[1])
+		}
+		if len(event.Topics) > 2 {
+			orderId = event.Topics[2]
+		}
+		switch event.Address {
+		case OrderBookContractAddress:
+			orderType = "limit"
+			switch event.Topics[0] {
+			case cep.orderBookABI.Events["OrderPlaced"].ID:
+				err := cep.orderBookABI.UnpackIntoMap(args, "OrderPlaced", event.Data)
+				if err != nil {
+					log.Error("error in orderBookABI.UnpackIntoMap", "method", "OrderPlaced", "err", err)
+					continue
+				}
+				eventName = "OrderPlaced"
+				order := LimitOrder{}
+				order.DecodeFromRawOrder(args["order"])
+				args["order"] = map[string]interface{}{
+					"ammIndex":          order.AmmIndex,
+					"trader":            order.Trader,
+					"baseAssetQuantity": utils.BigIntToFloat(order.BaseAssetQuantity, 18),
+					"price":             utils.BigIntToFloat(order.Price, 6),
+					"reduceOnly":        order.ReduceOnly,
+					"salt":              order.Salt,
+				}
+
```

Comment:
so this process is happening 2ice. once in `ProcessEvents` and again here. should try to remove duplication. same for other events.